shared/GIMessage.js

```diff
@@ -25,27 +25,27 @@ sbp('okTurtles.data/set', 'CHELONIA_CONFIG', {
 
 export class GIMessage {
   // flow type annoations to make flow happy
-  _message: Object
-
+  _decrypted: GIOpValue
   _mapping: Object
+  _message: Object
 
-  static OP_CONTRACT = 'c'
-  static OP_ENCRYPTED_ACTION = 'ea' // e2e-encrypted action
-  static OP_PUBLIC_ACTION = 'pa' // publicly readable action
-  static OP_KEY_AUTH = 'ka' // add this key to the list of keys allowed to write to this contract
-  static OP_KEY_UPDATE = 'ku' // update a key's context
-  static OP_KEY_DEAUTH = 'kd' // remove this key from authorized keys
-  static OP_PROTOCOL_UPGRADE = 'pu'
-  static OP_PROP_SET = 'ps' // set a public key/value pair
-  static OP_PROP_DEL = 'pd' // delete a public key/value pair
+  static OP_CONTRACT: 'c' = 'c'
+  static OP_ENCRYPTED_ACTION: 'ea' = 'ea' // e2e-encrypted action
+  static OP_PUBLIC_ACTION: 'pa' = 'pa' // publicly readable action
+  static OP_KEY_AUTH: 'ka' = 'ka' // add this key to the list of keys allowed to write to this contract
+  static OP_KEY_UPDATE: 'ku' = 'ku' // update a key's context
+  static OP_KEY_DEAUTH: 'kd' = 'kd' // remove this key from authorized keys
+  static OP_PROTOCOL_UPGRADE: 'pu' = 'pu'
+  static OP_PROP_SET: 'ps' = 'ps' // set a public key/value pair
+  static OP_PROP_DEL: 'pd' = 'pd' // delete a public key/value pair
```

Comment:
For these, since they might be adjusted in a future PR of mine, can you set their type to `GIOpType` instead? That way fewer replacements will need to be made.

Reply:
I'm going to try that. Let's hope Flow doesn't complain!

Reply:
Hum... Flow is asking me to put the literal annotations back, because these static values get later used
as computed properties:
![image](https://user-images.githubusercontent.com/64228468/105109446-d6430500-5abc-11eb-96a4-efcc1e6f378e.png)

Reply:
Jeez... does it really need the type annotations then? Can you simply remove them? (Is it that dumb that it needs them here?)

Reply:
A possible solution would be to use a language construct that avoids the need for computed property syntax, for example a switch statement (somewhat ugly), or building `opFns` using `Object.fromEntries()` (might need to be extracted so that it'd be only built once).

Reply:
@taoeffect If I remove these type annotations, it starts complaining about not being able to build a typed interface for that module :(

Reply:
Yeah we can keep the literals if that's the best way forward